### PR TITLE
sanitize + in KUBE_VERSION during gce upgrade

### PR DESCRIPTION
**Pull request by mikedanese**
#14674

### CODE REVIEW

**Comment by mikedanese** [comment]
@k8s-bot ok to test

**Comment by mikedanese** [comment]
```
oidc.TestOIDCDiscoverySecureConnection
```

File: cluster/gce/upgrade.sh [hunk 1]
```diff
@@ -173,7 +173,7 @@ function upgrade-nodes() {
 #   KUBELET_KEY_BASE64
 function prepare-node-upgrade() {
   echo "== Preparing node upgrade (to ${KUBE_VERSION}). ==" >&2
-  SANITIZED_VERSION=$(echo ${KUBE_VERSION} | sed s/"\."/-/g)
+  SANITIZED_VERSION=$(echo ${KUBE_VERSION} | sed s/"\."/-/g | sed s/\+/-/g)
```
Comment (ihmccreery) [inline cluster/gce/upgrade.sh]: I'd fix this with a single sed: `sed s/[\.\+]/-/g`.

**Comment by ihmccreery** [comment]
See comment.  Otherwise LGTM; will apply once comment is done.  Not sure what the failing test is.

**Comment by mikedanese** [comment]
@ihmccreery fixed!

**Pull request merged**